Explorer/Assets/DCL/InWorldCamera/CameraReelGallery/Components/PagedCameraReelManager.cs

```diff
@@ -0,0 +1,67 @@
+using Cysharp.Threading.Tasks;
+using DCL.InWorldCamera.CameraReelStorageService;
+using DCL.InWorldCamera.CameraReelStorageService.Schemas;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace DCL.InWorldCamera.CameraReelGallery.Components
+{
+    public class PagedCameraReelManager
+    {
+        public bool AllImagesLoaded { get; private set; } = false;
+
+        private readonly ICameraReelStorageService cameraReelStorageService;
+        private readonly string walletAddress;
+        private readonly int pageSize;
+        private readonly int totalImages;
+
+        private int currentOffset = 0;
+        private int currentLoadedImages = 0;
+
+        public PagedCameraReelManager(
+            ICameraReelStorageService cameraReelStorageService,
+            string wallet,
+            int totalImages,
+            int pageSize)
+        {
+            this.cameraReelStorageService = cameraReelStorageService;
+            this.walletAddress = wallet;
+            this.totalImages = totalImages;
+            this.pageSize = pageSize;
+        }
+
+
+        public async UniTask<Dictionary<DateTime, List<CameraReelResponseCompact>>> FetchNextPageAsync(CancellationToken ct)
+        {
+            CameraReelResponsesCompact response = await cameraReelStorageService.GetCompactScreenshotGalleryAsync(walletAddress, pageSize, currentOffset, ct);
+            currentOffset += pageSize;
+
+            currentLoadedImages += response.images.Count;
+            AllImagesLoaded = currentLoadedImages == totalImages;
+
+            Dictionary<DateTime, List<CameraReelResponseCompact>> elements = new ();
```

Comment:
which object(s) would you pool? Because `CameraReelResponseCompact` is generated by the parsing of the json response of the BE api call few lines above, so by the time I create the dictionary those object have already been created.
Or maybe you mean to reuse the same dictionary avoiding the creation of a new one every `FetchNextPageAsync` call?